packages/starlight/integrations/expressive-code/index.ts

```diff
@@ -0,0 +1,154 @@
+	 * for details of the supported theme formats.
+	 *
+	 * Starlight uses the dark and light variants of Sarah Drasner’s
+	 * [Night Owl theme](https://github.com/sdras/night-owl-vscode-theme) by default.
+	 *
+	 * If you provide at least one dark and one light theme, Starlight will automatically keep
+	 * the active code block theme in sync with the current site theme. Configure this behavior
+	 * with the [`useStarlightDarkModeSwitch`](#usestarlightdarkmodeswitch) option.
+	 *
+	 * Defaults to `['starlight-dark', 'starlight-light']`.
+	 */
+	themes?: ThemeObjectOrBundledThemeName[] | undefined
+	/**
+	 * When `true`, code blocks automatically switch between light and dark themes when the
+	 * site theme changes.
+	 *
+	 * When `false`, you must manually add CSS to handle switching between multiple themes.
+	 *
+	 * **Note**: When setting `themes`, you must provide at least one dark and one light theme
+	 * for the Starlight dark mode switch to work.
+	 *
+	 * Defaults to `true`.
+	 */
+	useStarlightDarkModeSwitch?: boolean | undefined;
+	/**
+	 * When `true`, Starlight's CSS variables are used for the colors of code block UI elements
+	 * (backgrounds, buttons, shadows etc.), matching the
+	 * [site color theme](/guides/css-and-tailwind/#theming).
+	 *
+	 * When `false`, the colors provided by the active syntax highlighting theme are used for
+	 * these elements.
+	 *
+	 * Defaults to `true` if the `themes` option is not set (= you are using Starlight's
+	 * default themes), and `false` otherwise.
+	 *
+	 * **Note**: When manually setting this to `true` with your custom set of `themes`, you must
+	 * provide at least one dark and one light theme to ensure proper color contrast.
+	 */
+	useStarlightUiThemeColors?: boolean | undefined;
+};
+
+export const starlightExpressiveCode = ({
+	astroConfig,
+	starlightConfig,
+	useTranslations,
+}: {
+	astroConfig: Pick<AstroConfig, 'root' | 'srcDir'>;
+	starlightConfig: StarlightConfig;
+	useTranslations: ReturnType<typeof createTranslationSystemFromFs>;
+}): AstroIntegration[] => {
+	const { locales, expressiveCode } = starlightConfig;
+	if (expressiveCode === false) return [];
+	const config: StarlightExpressiveCodeOptions =
+		typeof expressiveCode === 'object' ? expressiveCode : {};
+
+	const {
+		themes: themesInput,
+		customizeTheme,
+		styleOverrides: { textMarkers: textMarkersStyleOverrides, ...otherStyleOverrides } = {},
+		useStarlightDarkModeSwitch,
+		useStarlightUiThemeColors = config.themes === undefined,
+		plugins = [],
+		...rest
+	} = config;
+
+	// Handle the `themes` option
+	const themes = preprocessThemes(themesInput);
+	if (useStarlightUiThemeColors === true && themes.length < 2) {
+		console.warn(
+			[
+				`*** Warning: Using the config option "useStarlightUiThemeColors: true" `,
+				`with a single theme is not recommended. For better color contrast, `,
+				`please provide at least one dark and one light theme.\n`,
+			].join('')
```

Comment:
Prettier kept reformatting this on me, that's why I went with the array style. :) But you're right, this would be nicer! Let's just give your version a try. I didn't recheck after recently updating Prettier to see if it still reformats this.

Reply:
Yep, it works now! Nice!